cucumber-junit-platform-engine/src/main/java/io/cucumber/junit/platform/engine/DefaultNamingStrategyProvider.java

```diff
@@ -0,0 +1,92 @@
+package io.cucumber.junit.platform.engine;
+
+import io.cucumber.core.gherkin.Pickle;
+import io.cucumber.plugin.event.Node;
+import org.junit.platform.engine.ConfigurationParameters;
+
+import java.util.Locale;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static io.cucumber.junit.platform.engine.Constants.JUNIT_PLATFORM_LONG_NAMING_STRATEGY_EXAMPLE_NAME_PROPERTY_NAME;
+import static io.cucumber.junit.platform.engine.Constants.JUNIT_PLATFORM_SHORT_NAMING_STRATEGY_EXAMPLE_NAME_PROPERTY_NAME;
+
+enum DefaultNamingStrategyProvider {
+    LONG {
+        @Override
+        NamingStrategy create(ConfigurationParameters configuration) {
+            String exampleStrategy = configuration.get(JUNIT_PLATFORM_LONG_NAMING_STRATEGY_EXAMPLE_NAME_PROPERTY_NAME).orElse("example-number");
+            return createNamingStrategy(exampleStrategy,
+                    node -> longStrategy(node, nameOrKeyword(node)),
+                    (node, pickle) -> longStrategy(node, nameOrKeyword(node)),
+                    (node, pickle) -> longStrategy(node, pickle.getName()));
+        }
+    },
+
+    SHORT {
+        @Override
+        NamingStrategy create(ConfigurationParameters configuration) {
+            String exampleStrategy = configuration.get(JUNIT_PLATFORM_SHORT_NAMING_STRATEGY_EXAMPLE_NAME_PROPERTY_NAME).orElse("example-number");
+            return createNamingStrategy(exampleStrategy,
+                    DefaultNamingStrategyProvider::nameOrKeyword,
+                    (node, pickle) -> nameOrKeyword(node),
+                    (node, pickle) -> pickle.getName()
+            );
+        }
+    };
+
+    abstract NamingStrategy create(ConfigurationParameters configuration);
+
+    static DefaultNamingStrategyProvider getStrategyProvider(String name) {
+        return valueOf(name.toUpperCase(Locale.ROOT));
+    }
+
+    private static NamingStrategy createNamingStrategy(String exampleStrategy,
+                                                       Function<Node, String> name,
+                                                       BiFunction<Node, Pickle, String> exampleNameByNumber,
+                                                       BiFunction<Node, Pickle, String> exampleNameByPickle) {
+        BiFunction<Node, Pickle, String> exampleName;
```

Comment:
I think you can pull this out of the method and write something like:

```java
BiFunction<Node, Pickle, String> exampleName = configuration.get(JUNIT_PLATFORM_SHORT_NAMING_STRATEGY_EXAMPLE_NAME_PROPERTY_NAME)
  .map(parseExampleStrategy())
  .orElse(exampleNumberStrategy());
```  

Names are merely illustrative. So feel free to think of better ones.

This will give you some problems when trying to invoke the long strategy. And to solve those, perhaps it would help to think of the long naming strategy a function that generates a prefix based on the parent nodes while the short naming strategy always returns the empty string.